src/GenericAccessRequestForm/GenericAccessRequestForm.tsx

```diff
@@ -0,0 +1,317 @@
+            rules={[
+              {
+                type: 'email',
+                message: 'The input is not valid E-mail',
+              },
+              {
+                required: true,
+              },
+            ]}
+          >
+            <Input />
+          </Form.Item>
+          <Form.Item
+            name='Affiliated Institution'
+            label='Affiliated Institution'
+            rules={[
+              {
+                required: true,
+              },
+            ]}
+          >
+            <Input />
+          </Form.Item>
+          {specificFormInfo.showGrantNumber && (
+            <Form.Item
+              name='Grant Number'
+              label='Grant Number'
+              rules={[
+                {
+                  required: false,
+                },
+              ]}
+            >
+              <Input />
+            </Form.Item>
+          )}
+
+          <Form.Item
+            label='Role on Project'
+            name='Role on Project'
+            initialValue={role}
+            rules={[{ required: true, message: 'Please select a role' }]}
+          >
+            <Radio.Group onChange={onRadioChange} value={role}>
+              <Space direction='vertical'>
+                <Radio value={'Principal Investigator'}>
+                  Principal Investigator
+                </Radio>
+                <Radio value={'Co-Principal Investigator'}>
+                  Co-Principal Investigator
+                </Radio>
+                <Radio value={'Co-Investigator'}>Co-Investigator</Radio>
+                <Radio value={'Administrator'}>Administrator</Radio>
+                <Radio value={'Clinical Collaborator'}>
+                  Clinical Collaborator
+                </Radio>
+                <Radio value={'Clinical Coordinator'}>
+                  Clinical Coordinator
+                </Radio>
+                <Radio value={'Data Analyst'}>Data Analyst</Radio>
+                <Radio value={'Data Manager'}>Data Manager</Radio>
+                <Radio value={'Research Coordinator'}>
+                  Research Coordinator
+                </Radio>
+                <Radio value={'Other'}>
+                  Other...
+                  {role === 'Other' ? (
+                    <Form.Item
+                      name='Custom Role'
+                      noStyle
+                      rules={[
+                        { required: true, message: 'Please provide a role' },
+                      ]}
+                    >
+                      <Input style={{ width: 200, marginLeft: 8 }} />
+                    </Form.Item>
+                  ) : null}
+                </Radio>
+              </Space>
+            </Radio.Group>
+          </Form.Item>
+          <Form.Item {...tailLayout}>
+            <Space>
+              {!userHasAccess() ? (
+                <Tooltip
+                  title={`You don't have permission to
+                  request for access to study registration`}
```

Comment:
This contents need to be created dynamitically

Reply:
Notes from Chat with Mingfei: this should be updated to read
```You don't have permission to
                  request for access```

Reply:
This has now been updated to read
```You don't have permission to request for access```